Milestone-4 Additions

NickAspiras opened the pull request.

NickAspiras committed:
First Milestone

NickAspiras committed:
Milestone3Work!

NickAspiras committed:
Add files via upload

NickAspiras committed:
Create backend.md

NickAspiras committed:
Add files via upload

NickAspiras committed:
Delete index.js

NickAspiras committed:
Delete package.json

bbmcmann commented on milestone-3/backend/index.js:
```diff
@@ -0,0 +1,62 @@
+})
+
+app.get("/api/recipe/:recipeName", async (req, res) => {
+    const recipes = await Recipe.find({"/:recipeName"})
```
So when writing a mongoose query, the syntax can be a bit confusing. when you use the find() method, you want to find documents in your database that match whatever specifications you are looking for. The format for this is as follows:
find({ _fieldName_: **fieldValue** }). _fieldName_ would be whatever database document field you are trying to match. So in this case, you would probably want to use recipe_name since you are looking for a specific recipe based off its name.
**fieldValue** would be whatever specific value you are trying to find. So if you are looking for a taco recipe, this may be called 'tacos' or whatever it is in the database. This is the actual recipe name you are looking for.
Below is a link to a more detailed explanation of this. If this isn't clear enough and these recourses don't help feel free to reach out to me so I can help out! :)
https://www.notion.so/Writing-a-Query-b48e00b154544487af8df8a36acb1161

bbmcmann commented on milestone-3/backend/index.js:
```diff
@@ -0,0 +1,62 @@
+})
+
+app.post("/api/recipe", async (req, res) => {
+    const { recipe_name, recipe_description, imgsrc, id } = req.body
```
These values should match what you defined in your recipe schema. Currently you do not have an imgsrc or id defined so they need to be added to the schema.

bbmcmann commented on milestone-3/backend/index.js:
```diff
@@ -0,0 +1,62 @@
+app.put("/:recipeName/instruction", async(req, res) => {
+    const recipeName = req.params.recipeName;
+    const instruction = req.body.newInstruction;
+})
```
Both these put requests are incomplete. What you will need to do is find the recipe document in your database that you are trying to add an ingredient or instruction to. Then similar to you post, use a try catch statement to add the ingredient to your document and save, or display an error code.
You will probably want to use the mongoose push method (linked below) to add ingredients to array fields in your document. Again if you need any extra help I'm happy to give some advice!
https://mongoosejs.com/docs/api/array.html#mongoosearray_MongooseArray-push

bbmcmann commented on milestone-3/backend/index.js:
```diff
@@ -0,0 +1,62 @@
+        res.status(500).send(error.message)
+        console.log(`error is ${error.message}`)
+    }
+})
```
The rest of this post request looks good 👍

bbmcmann commented on milestone-3/backend/index.js:
```diff
@@ -0,0 +1,62 @@
+const Recipe = new mongoose.Schema({
+    recipe_name: String,
+    recipe_description: String
+}, {collection: 'Recipes'})
```
A few notes on this schema:
To help keep track of naming I may suggest using RecipeSchema and Recipe instead of Recipe and Setup respectively. Setup is a bit harder to follow in the code since it is not super descriptive.

When setting up schemas, it helps to think about what we want our actual documents to look like. Here you have defined the name and description fields in a document, but there should be other data we need to keep track of with recipes. Things such as ingredient list and an instruction list. Make sure all the data you want to store is represented in your schema.

Finally, this is just a small comment, but you should stick with camel case or snake case throughout your code when naming variables. Just something that will help with readability in project teams.

bbmcmann reviewed (commented):
Hey Nick, a lot of what you have is correct and on the way to being done. You have most of the groundwork setup and a few endpoints are good to go, but a couple others need a few more things to be complete. I added notes where all the changes need to be made, if you have any questions at all feel free to reach out! :)

NickAspiras committed:
Fixed files for Milestone-4

Added in changes and edits needed for Milestone 4

NickAspiras committed:
Merge pull request #1 from NickAspiras/NickAspiras-Milestone-4-edit

Fixed files for Milestone-4

NickAspiras committed:
Milestone-5 Work (I Think)

Made necessary changes to the frontend to connect react with backend, using useEffect and useState respectively. Additionally, I added in the routes to communicate with the backend.

NickAspiras committed:
Merge pull request #2 from NickAspiras/NickAspiras-Milestone-5-Work

Milestone-5 Work (I Think)

bbmcmann commented on milestone-3/frontend/src/App.js:
```diff
@@ -0,0 +1,58 @@
+import './App.css';
+import recipes from './components/data';
+import Title from './components/begin';
```
You import this Title component but never use it

bbmcmann commented on milestone-3/frontend/src/App.js:
```diff
@@ -0,0 +1,58 @@
+import Title from './components/begin';
+import Recipe from './components/Recipes';
+import Navbar from './components/navbar';
+import React, { useState, useState} from "react";
```
This give a compilation error

bbmcmann commented on milestone-3/frontend/src/App.js:
```diff
@@ -0,0 +1,58 @@
+          <div> 
+            <Navbar />
+            <div className = "pageData"> 
+              <Routes>
```
I believe this should be Switch instead of Routes
see: 
https://www.notion.so/Using-React-Router-c1365e4e3d9e46c8b04325063ce63da5

bbmcmann commented on milestone-3/frontend/src/components/Recipes.jsx:
```diff
@@ -0,0 +1,24 @@
+export default function Recipe(props) {
+    const [ingredient, setNewIngredient] = React.useState('');
+    const [instruction, setNewInstruction] = React.useState('');
+    const [ingredient, setNewIngredient] = useState('');
```
You have 2 state variables with the same names

bbmcmann reviewed (commented):
Hey Nick, sorry for the late review! It is sometimes hard to see when new commits are added to an existing PR on Git so I didn't see your additions earlier.

On my end your code does not even compile, I went through it a bit and tried to edit some of these compilation errors in case they were just some small unsaved changes. There ended up being quite a few and I couldn't get it to run without making a ton of edits. I noted some of these errors in the review.

I'm not sure if this is just some weird bug, or maybe the files you wanted didn't upload. Does it compile and run correctly on your end? If you review this PR and notice that the wrong files uploaded or something just make a new one with the correct ones. Otherwise, we will have to figure out why it won't even compile on this end.

Either way, I am going to close this PR because your milestone 4 stuff looks good, and once the issues are found you can make a new milestone 5 PR. If you have questions or need any help let me know!
